### ISSUE TITLE
Suggestion for Dark Mode Option

### ISSUE THREAD

**Issue by N8Solutions** [gantry/gantry5#2595]
I'd like to suggest that an option for Dark Mode be added to the Gantry 5 Templates. More and more Apps have a Dark Mode setting or are able to recognize if Dark Mode is enabled on the device and will then enable Dark Mode automatically in the app.

Dark Mode is basically just a different color scheme so there could be multiple ways to implement this. I've seen some a couple of extensions for this but they seem very basic and don't provide a lot of functionality or options.

My idea would be to add a Toggle Switch in the admin area either under the "Styles" tab or the "Page Settings" tab for "Enabling" Dark Mode options globally throughout the Theme in the Base Outline and of course could allow for it to be disabled as an override in other Outlines.

Under the "Styles" tab, each section should have an extra set of color options for Dark Mode.

A Particle, and/or an Atom, for Enabling/Disabling Dark Mode should be available for the Front End so the site visitor can choose to enable or disable Dark Mode as well as allow the site designer to choose where they would like to place the Toggle for Dark Mode.

I do not know if this next part is possible or not, but it would be nice if, when Dark Mode is enabled on the device being used to browse the site, that the site automatically recognizes this and shows the Dark Mode style instead of the Light Mode.

It should also be noted that some browsers will detect Dark Mode and will show their interface in Dark Mode when Dark Mode is enabled on the device. Chrome, Safari, & Firefox all do this but the website content is not modified, only the browser interface is. Firefox goes one step further adding an option for what it calls "Night Mode" which will invert the colors of the websites being viewed. I'm not sure if it is possible to override Firefox's "Night Mode" or not for websites with Dark Mode enabled.

Another thought regarding the color styles, I realize it would be difficult to back port such an option to so many Gantry 5 temples.
So instead of adding extra color options under the Styles tab, what about

- When Dark Mode is enabled it creates a second Styles tab in the Outlines to be used with Dark Mode and the colors in the second Styles tab are Automatically inverted the first time it's turned on. Then the colors could be customized so a more appropriate Dark Theme could be created.

- Similar to the above method, another option would be to add an "Association" between two different Outlines where one is marked as Light Mode and the other as Dark Mode, and which one gets used depends on whether or not Dark Mode is enabled.

I'm making this suggestion because I'm seeing Dark Mode being made available in more apps as time goes by. Yesterday morning Google's John Mueller tweeted the following reply to another:
> In general, I'd recommend making sure that your hosting is secure, automatically updated, and multi-zoned if you want your site to rank well at the end of the universe. ***Dark Mode is also a plus***.

![29AE03A9-6431-4B4D-9716-78D3BD58EA7E](https://user-images.githubusercontent.com/12389693/73164485-b6237480-40fa-11ea-849a-0146b37a14ae.jpeg)

[https://twitter.com/johnmu/status/1220816647273009152?s=21](https://twitter.com/johnmu/status/1220816647273009152?s=21)

This tweet indicates that having a Dark Mode option for your site is a ranking factor that Google takes into consideration. We don't know how much weight Google gives this ranking factor but every little bit helps.

Maybe I'm over complicating this. If you have a better idea or a solution that you like and are already using please let me know.

**Comment by sanek4life** [comment]
@N8Solutions  I completely agree with you and I also suggested introducing such a regime into the core of the CMS Joomla.

I hope at least someone already does this.

See all the details here - #26772

**Comment by N8Solutions** [comment]
@sanek4life I'm not sure why but the link you provided does not work if clicked on. If you copy and paste, it works just fine. I've corrected the link so this one should work.
[https://github.com/joomla/joomla-cms/issues/26772](https://github.com/joomla/joomla-cms/issues/26772)
Thanks for that by the way.

I'll take some time and look over this link that was mentioned in the other thread.
[https://web.dev/prefers-color-scheme/](https://web.dev/prefers-color-scheme/)

I also really like how the [Symfony website](https://symfony.com/) changes the themes without having to reload the page. I also tried it out on my phone with the phone set Dark Mode and it recognized that the phone was in Dark Mode and showed the Dark Theme of the website.

Another site mentioned was [https://dri.es/](https://dri.es/). Their site didn't have a button/menu item to turn on a dark theme but when you visit the website with your phone in Dark Mode, just like the Symfony website, it automatically shows the Dark theme.

As a test, I disabled Dark Mode on my phone and then reloaded the home page on both of those sites and they automatically switched back to Light Mode.

This would be really great for Gantry 5 templates if this was implemented!

**Comment by mahagr** [comment]
To be honest, I don't think this is a Gantry feature, but a feature in individual Gantry themes. There is nothing in Gantry itself required to make it work in a theme.

**Comment by N8Solutions** [comment]
> To be honest, I don't think this is a Gantry feature, but a feature in individual Gantry themes. There is nothing in Gantry itself required to make it work in a theme.
> 
> Also... You need to add support to every single theme to make this to work. And this is even if the base theme had the support for the feature.

@mahagr I completely agree. That's why I suggested **Dark Mode be added to the Gantry 5 Templates**. It would be great if this started with the Hydrogen and Helium templates then worked in to the paid templates at RT.
Of course maybe other template designers like @JoomFX would like to implement this as well.

I've been looking over this link [https://web.dev/prefers-color-scheme/](https://web.dev/prefers-color-scheme/), and while I think it would be a great addition to the Gantry 5 themes if this was available natively, it doesn't look too difficult to add. It mostly appears to be time consuming to create a color scheme for Dark Mode.

Maybe this is also something that could be done as an **Atom** to be added to Page Settings. A **Dark Mode Atom** would definitely make it easier to add the Dark Mode feature across all the Gantry 5 Templates.

So what do you think @mahagr?
If it were to be directly integrated in to the templates, my thoughts would be to do it this way.
- add a toggle switch to make dark mode available
- when the toggle is switched on to make dark mode available, it enables a 2nd "Styles" tab for dark mode customization
- add another toggle switch to enable automatic detection and switching to dark mode if the browser is in dark mode
- create a Particle that adds a toggle switch to the front end of the website to allow the site visitor to manually switch between light and dark mode.

As far as implementing something like this as an Atom, I'm sure it could be done similarly within the Atom. @thexmanxyz thoughts?

**Comment by sanek4life** [comment]
@mahagr I completely agree with the comment above, this can be done as a particle in Gantry 5, this should be a switch from a standard theme to a dark theme.

I see no reason to renew my subscription by RocketTheme if I don't see templates with a **dark theme**!

**Comment by sanek4life** [comment]
I found some solution here - [https://github.com/twbs/bootstrap/issues/27514#issuecomment-508972071](https://github.com/twbs/bootstrap/issues/27514#issuecomment-508972071)

I hope that it will be possible to add a dark theme to each @Gantry 5 template.

**Comment by marcomueller68** [comment]
I tried to do the following as suggested by copilot but it didn't change the background:
Step-by-Step Solution

    Verify Custom CSS and JavaScript Files:
        Ensure the custom CSS and JavaScript files are placed in the correct directories.
        Verify the contents of these files.

    Edit the gantry5.php File:
        Ensure the custom CSS and JavaScript files are included correctly.

    Debugging Steps:
        Use the browser's developer tools to check if the custom files are loaded.
        Verify if the styles are being overridden by other styles.

Step 1: Verify Custom CSS and JavaScript Files
Custom CSS File

Ensure the custom CSS file is placed in the css-compiled directory:
/* Custom styles for Gantry 5 Admin Interface */
body.g-admin .g5-container,
body.g-admin .g5-container .g-content,
body.g-admin .g-grid,
body.g-admin .g-grid > .g-block {
    background-color: #1e1e1e !important;
    color: #dcdcdc !important;
}

body.g-admin .g-grid .g-block .g-content {
    background-color: #1e1e1e !important;
    color: #dcdcdc !important;
}

/* Add more specific styles as needed */

Custom JavaScript File

Ensure the custom JavaScript file is placed in the appropriate directory:
document.addEventListener('DOMContentLoaded', function() {
    const gantryContainer = document.querySelector('.g5-container');

    if (gantryContainer) {
        gantryContainer.style.backgroundColor = '#1e1e1e';
        gantryContainer.style.color = '#dcdcdc';
    }

    const gridContainer = document.querySelector('.g-grid');
    if (gridContainer) {
        gridContainer.style.backgroundColor = '#1e1e1e';
        gridContainer.style.color = '#dcdcdc';
    }
});

Step 2: Edit the gantry5.php File to Load Custom Files

Ensure the custom CSS and JavaScript files are included correctly:
<?php
// Other PHP code...

// Ensure custom CSS is loaded last
echo '<link rel="stylesheet" type="text/css" href="/administrator/components/com_gantry5/css-compiled/custom-gantry-admin.css">';

// Ensure custom JavaScript is loaded last
echo '<script src="/administrator/components/com_gantry5/js/custom-gantry-admin.js"></script>';

// Other PHP code...
?>

Step 3: Debugging Steps

    Use Browser's Developer Tools:
        Open the browser's developer tools (usually by pressing F12).
        Go to the "Network" tab and reload the page to see if your custom CSS and JavaScript files are being loaded.
        If the files are not loaded, check for any errors in the "Console" tab.

    Check for Style Overrides:
        In the "Elements" tab of the developer tools, inspect the elements that should have the dark background.
        Check the "Styles" pane to see if your custom styles are being overridden by other styles.

Summary:

    Verify Custom CSS and JavaScript Files: Ensure they are placed in the correct directories and contain the correct content.
    Edit the gantry5.php File: Ensure the custom CSS and JavaScript files are included correctly.
    Debugging Steps: Use the browser's developer tools to check if the custom files are loaded and if the styles are being overridden.

By following these steps, you should be able to enforce your custom dark theme styles in the Gantry 5 layout editor. If you encounter any issues, feel free to ask for further assistance!
